### ISSUE TITLE
It takes multiple seconds for the first `localStorage.getItem` call

### ISSUE THREAD

**Issue by fabiospampinato** [electron/electron#24441]
### Issue Details

* **Electron Version:**
  * 9.1.0
* **Operating System:**
  * macOS 10.14
* **Last Known Working Electron version:**
  * Maybe Electron <= v7, although maybe I wasn't using localStorage at all back then 🤔 

### Expected Behavior

LocalStorage shouldn't be absurdly slow.

### Actual Behavior

LocalStorage is absurdly slow, requiring about 5 seconds on my machine for this call: 

```
localStorage.getItem('');
```

### To Reproduce

This is a weird one, I spent a couple of hours tracking down this one, you might be able to reproduce it with the files provided but I'm not sure.

1. Running `npx electron app_dummy_slow.asar` requires about 5 seconds on my machine for a window to show up.
2. No need really to unpack the asar, it's essentially the quick-start project with only 2 interesting lines, without either of which I can't reproduce the issue:
   1. `localStorage.getItem('');`
      - Putting this call somewhere in `index.html` takes about 5 seconds to complete, which sounds absurd to me. Without this call I can't experience any issues.
      - The same thing happens when using `sessionStorage`.
   2. `"name": "notable",`
      - This is in the `package.json` file, using a different name I can't reproduce the issue. As "notable" is the name of my Electron app I guess when setting this name its specific local storage and session storage files are getting loaded, which for whatever reason takes forever.
3. The "Local Storage" and "Session Storage" folders, attached below, found under "~/Library/Application Support/Notable", weigh like less than 50kb each, making this even weirder.
4. Somewhat surprisingly Notable's appData folder weighs about 20MB on my machine, while the analogous folders for other electron apps on my machine weigh about 500kb each, but its content seems ok to me:

![image](https://user-images.githubusercontent.com/1812093/86703511-dd686100-c00b-11ea-8e0b-f4585fd9caf0.png)

### Resources

[app_dummy_slow.asar.zip](https://github.com/electron/electron/files/4882281/app_dummy_slow.asar.zip)
[Session Storage.zip](https://github.com/electron/electron/files/4882306/Session.Storage.zip)
[Local Storage.zip](https://github.com/electron/electron/files/4882307/Local.Storage.zip)

**Comment by fabiospampinato** [comment]
One of my users who can also experience this issue has a ~650kb app data folder, even loading all that from spinning rust shouldn't take that 5 seconds I think.

**Comment by fabiospampinato** [comment]
I've just measured some times and it actually takes about ~3750ms for that call alone on my machine, and like 50% of the times the number is within 5~10ms of that, almost like if there was a timeout of 3700ms somewhere or something.

**Comment by fabiospampinato** [comment]
I think this is probably a duplicate of #22438, as I can reproduce how having only one instance open of the app makes this issue disappear

**Comment by lishid** [comment]
I've ran into this issue in the past as well. My hypothesis was that chromium wasn't designed for multiple instances to run on top of the same folder, and the two instances run into some kind of lock contention reading/writing those files. They also don't "synchronize" across multiple running instance and end up with data from one of the two instances.

I've since worked around the problem by using a single main instance to manage multiple renderers, using `app.requestSingleInstanceLock()`. I've also been avoiding the use of `localStorage` and rely on JSON file instead.
